src/convert.rs

```diff
@@ -0,0 +1,53 @@
+use bdk::bitcoin::consensus::Decodable;
+use bitcoin::consensus::Encodable;
+
+pub fn bdk_block_hash_to_bitcoin_block_hash(hash: bdk::bitcoin::BlockHash) -> bitcoin::BlockHash {
+    use bdk::bitcoin::hashes::Hash as _;
+    let bytes = hash.as_byte_array().to_vec();
+
+    use bitcoin::hashes::sha256d::Hash;
+    use bitcoin::hashes::Hash as _;
+    let hash: bitcoin::hashes::sha256d::Hash = Hash::from_slice(&bytes).unwrap();
+
+    bitcoin::BlockHash::from_raw_hash(hash)
+}
+
+pub fn bitcoin_tx_to_bdk_tx(
+    tx: bitcoin::Transaction,
+) -> Result<bdk::bitcoin::Transaction, bdk::bitcoin::consensus::encode::Error> {
+    let mut tx_bytes = vec![];
+    tx.consensus_encode(&mut tx_bytes)
+        // Not really going to happen, mostly to satisfy the compiler.
+        .map_err(|err| {
+            bdk::bitcoin::consensus::encode::Error::Io(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                err.to_string(),
+            ))
+        })?;
+
+    let decoded = bdk::bitcoin::Transaction::consensus_decode(&mut tx_bytes.as_slice())?;
+
+    Ok(decoded)
+}
+
+pub fn bdk_txid_to_bitcoin_txid(hash: bdk::bitcoin::Txid) -> bitcoin::Txid {
+    use bdk::bitcoin::hashes::Hash as _;
+    let bytes = hash.as_byte_array().to_vec();
```

Comment:
You don't need to convert to a `Vec<u8>` here, which will allocate unnecessarily.
You can do
```rust
use bitcoin::hashes::Hash as _;
use bdk::bitcoin::hashes::Hash as _;

bitcoin::Txid::from_byte_array(txid.to_byte_array())
```